examples/hello_malloc.c

```diff
@@ -0,0 +1,73 @@
+#include <stdint.h>
```

Comment:
These two files currently play the role of both tests of malloc/new and examples for their usage.

As tests, they could be improved by incorporating them into our end-to-end test suite, replacing the printing with appropriate failure signaling (typically, `abort()`ing if anything goes wrong). This would allow us to get automatically notified if we break malloc.

As examples, they could be improved by simplifying the remainder down into something very simple: mallocing some memory, doing something with it, then freeing it. This would make it easier to quickly evaluate the runtime and code overhead of malloc by examining the generated output, which would help users understand whether or not it's a good fit for their designs.

Reply:
These have been simplified, and the "real tests" have been added to the test_suite PR.  To make this work I defined an __LLVM_MOS_SDK preprocessor #define so the tests could properly exercise the failed allocation paths and the heap limit functionality.